### PR TITLE
Adds unit test for PageHeader component

### LINKED ISSUE #1337: dashboard components require further unit testing code coverage
In order to further improve the automated unit testing workflow for pbench dashboard, greater code coverage of dashboard components is required for catching all possible scenarios. By striving for complete code coverage, we can regularly maintain quality of tests, refactor components with confidence, and catch errors before deploying to production. 

Coverage reports are currently retrieved in Jest by running the following command: `umi test ./src/components ./src/pages --verbose --maxWorkers=1 --runInBand "--coverage"`

```
---------------------------------|----------|----------|----------|----------|-------------------|
File                             |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------------------------|----------|----------|----------|----------|-------------------|
All files                        |    18.87 |     4.99 |     9.82 |     18.9 |                   |
 components/Button               |    33.33 |      100 |        0 |    33.33 |                   |
  index.js                       |    33.33 |      100 |        0 |    33.33 |               5,7 |
 components/MonthSelect          |       20 |        0 |        0 |       20 |                   |
  index.js                       |       20 |        0 |        0 |       20 |... 42,44,53,54,59 |
 components/PageHeader           |     7.35 |     2.63 |     5.88 |     7.46 |                   |
  index.js                       |     7.35 |     2.63 |     5.88 |     7.46 |... 11,218,225,250 |
 components/RowSelection         |       40 |        0 |        0 |       40 |                   |
  index.js                       |       40 |        0 |        0 |       40 |          19,20,22 |
 components/SearchBar            |    15.79 |        0 |        0 |    15.79 |                   |
  index.js                       |    15.79 |        0 |        0 |    15.79 |... 40,42,54,55,62 |
 components/SiderMenu            |    16.25 |     4.26 |       20 |    15.79 |                   |
  SiderMenu.js                   |    16.25 |     4.26 |       20 |    15.79 |... 13,214,215,217 |
 components/Table                |    33.33 |      100 |        0 |    33.33 |                   |
  index.js                       |    33.33 |      100 |        0 |    33.33 |               5,7 |
 components/TableFilterSelection |     5.26 |        0 |        0 |     5.41 |                   |
  index.js                       |     5.26 |        0 |        0 |     5.41 |... 97,106,109,131 |
 components/TimeseriesGraph      |    18.18 |        0 |        0 |    18.18 |                   |
  index.js                       |    18.18 |        0 |        0 |    18.18 |... 48,50,55,66,68 |
 components/_utils               |      100 |      100 |      100 |      100 |                   |
  pathTools.js                   |      100 |      100 |      100 |      100 |                   |
 layouts                         |        0 |        0 |        0 |        0 |                   |
  PageHeaderLayout.js            |        0 |        0 |        0 |        0 |                 7 |
 pages/ComparisonSelect          |    23.91 |        0 |    11.11 |    23.91 |                   |
  index.js                       |    23.91 |        0 |    11.11 |    23.91 |... 38,140,142,167 |
 pages/Controllers               |    26.67 |        0 |     9.09 |    26.67 |                   |
  index.js                       |    26.67 |        0 |     9.09 |    26.67 |... 46,152,176,178 |
 pages/Results                   |    26.19 |        0 |    11.76 |    26.19 |                   |
  index.js                       |    26.19 |        0 |    11.76 |    26.19 |... 37,148,177,179 |
 pages/RunComparison             |    23.01 |     7.69 |     6.98 |    23.21 |                   |
  index.js                       |    23.01 |     7.69 |     6.98 |    23.21 |... 65,568,601,603 |
 pages/Search                    |    27.78 |     8.33 |    11.11 |    27.78 |                   |
  index.js                       |    27.78 |     8.33 |    11.11 |    27.78 |... 69,272,303,305 |
 pages/Summary                   |    40.74 |    46.15 |    18.18 |    40.74 |                   |
  index.js                       |    40.74 |    46.15 |    18.18 |    40.74 |... 62,176,190,223 |
 utils                           |    14.15 |     3.52 |     7.46 |    14.47 |                   |
  parse.js                       |    19.28 |     5.56 |    11.11 |    19.27 |... 79,489,496,503 |
  utils.js                       |     2.94 |        0 |        0 |     3.23 |... 90,191,193,195 |
---------------------------------|----------|----------|----------|----------|-------------------|
```

### PR DESCRIPTION
**Pull request by Anisha1234**
Fixes #1337 
Code Coverage Report:
``` 
components/PageHeader           |    98.53 |    69.74 |      100 |    98.51 |                   |
  index.js                      |    98.53 |    69.74 |      100 |    98.51 |                15 |
```

Coverage reports are currently retrieved in Jest by running the following command: `umi test ./src/components ./src/pages --verbose --maxWorkers=1 --runInBand "--coverage"`

### CODE REVIEW

**Comment by Anisha1234** [comment]
@gurbirkalsi I am not being able to test a single line i.e line no-15. It would be really helpful if you can give any insights about how to test the line as I have already tried testing the return result with every possible arguments.

**Comment by gurbirkalsi** [comment]
@Anisha1234 We can omit line 15 from test coverage as a function referenced from an external library is being invoked here. In general, we expect libraries used with the dashboard to maintain their own tests.

File: web-server/v0.4/src/components/PageHeader/index.test.js [hunk 1]
```diff
@@ -10,6 +13,21 @@ const routerData = {
   },
 };
 
+const { TabPane } = Tabs;
+const mockProps = {
+  selectedControllers: undefined,
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.test.js]: Can we consider initializing `selectedControllers` as the same default data type/structure we specify in the Redux store?

File: web-server/v0.4/src/components/PageHeader/index.test.js [hunk 2]
```diff
@@ -19,4 +37,72 @@ describe('test getBreadcrumb', () => {
...
+    const conversionFromLocation = jest.spyOn(wrapper.instance(), 'getBreadcrumbProps');
+
+    wrapper2.setProps({ selectedControllers: ['a', 'b', 'c'] });
+    getBreadcrumbDom();
+    expect(wrapper.state('breadcrumb')).toEqual(conversionBreadcrumbList());
+    expect(conversionBreadcrumbList).toHaveBeenCalled();
+
+    expect(getBreadcrumbProps).toHaveBeenCalled();
+
+    // eslint-disable-next-line no-unused-vars
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.test.js]: We can omit the unused variables from the destructured reference and remove the disabled eslint line here to improve readability.

File: web-server/v0.4/src/components/PageHeader/index.test.js [hunk 3]
```diff
@@ -19,4 +37,72 @@ describe('test getBreadcrumb', () => {
...
+    const breadcrumb = breadcrumbNameMap[url];
+    expect(getBreadcrumb(breadcrumbNameMap, url)).toBe(breadcrumb);
+  });
+});
+
+describe('test rendering of TableFilterSelection page component', () => {
+  it('render with empty props', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('Check functions implementation', () => {
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.test.js]: We should consider improving the name of the test to further reflect which function/behavior is being tested.

File: web-server/v0.4/src/components/PageHeader/index.test.js [hunk 4]
```diff
@@ -19,4 +37,72 @@ describe('test getBreadcrumb', () => {
...
+  it('check rendering and change simulation', () => {
+    expect(wrapper.find(Tabs).length).toEqual(1);
+    expect(wrapper.find(TabPane)).toHaveLength(2);
+    wrapper
+      .find(Tabs)
+      .at(0)
+      .simulate('change', 'key1');
+    expect(wrapper.instance().props.onTabChange).toHaveBeenCalled();
+  });
+  it('test the titleHeader', () => {
```
Comment (gurbirkalsi) [inline web-server/v0.4/src/components/PageHeader/index.test.js]: For example, naming this test 'should render selected controllers in titleheader' describes the behavior of this test case clearly by letting us know what specific behavior is expected.

**Comment by portante** [comment]
@Anisha1234, is this work still a `WIP`?

**Comment by Anisha1234** [comment]
> @Anisha1234, is this work still a `WIP`?

@portante I am blocked in this PR. Not being able to resolve the issue mentioned in the prev [comment](https://github.com/distributed-system-analysis/pbench/pull/1393#issuecomment-546349178).

**Pull request merged**